rampwf/prediction_types/detection.py

```diff
@@ -22,11 +22,11 @@ def __init__(self, y_pred=None, y_true=None, n_samples=None,
         if y_pred is not None:
             if fold_is is not None:
                 y_pred = y_pred[fold_is]
-            self.y_pred = np.array(y_pred)
+            self.y_pred = np.array(y_pred, dtype=object)
         elif y_true is not None:
             if fold_is is not None:
                 y_true = y_true[fold_is]
-            self.y_pred = np.array(y_true)
```

Comment:
I'm not 100% confident this is the right fix, but the issue seems to be that for detection prediction_types the input y_true/y_pred can be some ragged array of the form,
```py
np.array([[(1, 2, 3)], None])
```
which no longer works latest numpy without dtype=object

At the same time, I don't see what else we could do with such input. The other alternative could be,
```py
self.y_pred = np.zeros(len(y_true), dtype=object)
for idx, row in enumerate(y_true):
	self.y_pred[idx] = row
```
but it feels more brittle.